### PR TITLE
Added sharepoint connector

### PR DESCRIPTION
**Pull request by hagen6835**

### CODE REVIEW

File: backend/celerybeat-schedule
Comment (yuhongsun96) [inline backend/celerybeat-schedule:1]: Why is this needed/added?

File: web/src/components/icons/icons.tsx [hunk 1]
```diff
@@ -50,6 +50,7 @@ import hubSpotIcon from "../../../public/HubSpot.png";
 import document360Icon from "../../../public/Document360.png";
 import googleSitesIcon from "../../../public/GoogleSites.png";
 import zendeskIcon from "../../../public/Zendesk.svg";
+import sharepointIcon from "../../../public/RequestTracker.png";
```
Comment (yuhongsun96) [inline web/src/components/icons/icons.tsx]: Hm... suspicious

File: backend/danswer/connectors/sharepoint/connector.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+                client_id=aad_client_id,
+                client_credential=aad_client_secret
+            )
+            token = app.acquire_token_for_client(scopes=["https://graph.microsoft.com/.default"])
+            return token
+        
+        self.graph_client = GraphClient(_acquire_token_func)
+        return None
+    
+    def _prune_driveitem_list_by_time(
```
Comment (yuhongsun96) [inline backend/danswer/connectors/sharepoint/connector.py]: Is there a way to add a time range to the query so we don't need to process all drive items through this?

File: backend/danswer/connectors/sharepoint/connector.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+        
+        return pruned_driveitem_object_list
+    
+    def get_all_driveitem_objects(self, site_object_list: list[Site]) -> list[DriveItem]:
+        driveitem_list = []
+        for site_object in site_object_list:
+            site_list_objects = site_object.lists.get().execute_query()
+            for site_list_object in site_list_objects:
+                try:
+                    driveitems = site_list_object.drive.root.get_files(True).execute_query()
```
Comment (yuhongsun96) [inline backend/danswer/connectors/sharepoint/connector.py]: Like here, there's probably some filter options right?

File: backend/danswer/connectors/sharepoint/connector.py [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+            self, driveitem_object:DriveItem, 
+    ) -> Document:
+        file_text = self.extract_driveitem_text(driveitem_object)
+        doc = Document(
+            id=driveitem_object.id, 
+            sections=[Section(link=driveitem_object.web_url, text=file_text)],
+            source=DocumentSource.SHAREPOINT,
+            semantic_identifier=driveitem_object.name,
+            doc_updated_at=driveitem_object.last_modified_datetime.replace(tzinfo=timezone.utc), 
+            primary_owners=[BasicExpertInfo(display_name=driveitem_object.last_modified_by.user.displayName,
```
Comment (yuhongsun96) [inline backend/danswer/connectors/sharepoint/connector.py]: Is this guaranteed to exist?

File: backend/danswer/connectors/sharepoint/connector.py [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+                                            email=driveitem_object.last_modified_by.user.email)],
+            metadata={},
+        )
+        return doc
+
+    def extract_driveitem_text(self, driveitem_object:DriveItem) -> str:
+        driveitem_name =  driveitem_object.name
+        driveitem_text = UNSUPPORTED_FILE_TYPE_CONTENT
+
+        if driveitem_name.endswith('.docx') :
```
Comment (yuhongsun96) [inline backend/danswer/connectors/sharepoint/connector.py]: Let's also include doc and notes.  Though perhaps notes isn't a "file" type but let's make sure to include those. Also let's just make a note in comment here that powerpoints are useful but just point out the complexity why it's not trivial to add here

File: web/src/app/admin/connectors/sharepoint/page.tsx [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+            <CredentialForm<SharepointCredentialJson>
+              formBody={
+                <>
+                  <TextFormField
+                    name="aad_client_id"
+                    label="Azure AD Client ID:"
+                  />
+                  <TextFormField
+                    name="aad_client_secret"
+                    label="Azure AD Client Secret:"
```
Comment (yuhongsun96) [inline web/src/app/admin/connectors/sharepoint/page.tsx:116]: Slight preference to have secret at the end. Probably best in general to have most generic to most sensitive. So Directory ID, Client ID, then Client Secret

**Review by yuhongsun96 — COMMENTED** [review]
Looks good, minor comments. More importantly, let's be sure to support doc and notes

File: backend/celerybeat-schedule
Comment (hagen6835) [reply backend/celerybeat-schedule:1]: its not, we should remove it

File: web/src/components/icons/icons.tsx (hunk 1 above)
Comment (hagen6835) [reply web/src/components/icons/icons.tsx]: Fixed that

File: web/src/app/admin/connectors/sharepoint/page.tsx (hunk 6 above)
Comment (hagen6835) [reply web/src/app/admin/connectors/sharepoint/page.tsx:116]: fixed

File: backend/danswer/connectors/sharepoint/connector.py (hunk 5 above)
Comment (hagen6835) [reply backend/danswer/connectors/sharepoint/connector.py]: we should discuss this more

File: backend/danswer/connectors/sharepoint/connector.py (hunk 3 above)
Comment (hagen6835) [reply backend/danswer/connectors/sharepoint/connector.py]: i can do something like this 
```
driveitems = site_list_object.drive.root.get_files(True).order_by("last_modified_datetime").top(100).execute_query()
```
which would grab the last 100 modified documents. however there's no guarantee that more or less than 100 documents have been modified since the last indexing

File: backend/danswer/connectors/sharepoint/connector.py (hunk 4 above)
Comment (hagen6835) [reply backend/danswer/connectors/sharepoint/connector.py]: pretty certain a document always has a user attached to it or it will list "anon"

File: web/public/Sharepoint.png
Comment (yuhongsun96) [inline web/public/Sharepoint.png:1]: Wait I just realized this is the outdated logo, can you use the new one? Also the aspect ratio is a bit messed up on the frontend, should be a quick fix

**Comment by yuhongsun96** [comment]
Aight, good stuff, putting it in
